Makefile

```
@@ -297,120 +311,137 @@ $(O_FILES): | asset_files
 
 .PHONY: o_files asset_files
 
+
 #### Main commands ####
 
 ## Cleaning ##
 clean:
-	$(RM) -rf $(ROMC) $(ROM) $(ELF) build
+	$(RM) -r $(BUILD_DIR)
 
 assetclean:
-	$(RM) -rf $(ASSET_BIN_DIRS)
-	$(RM) -rf assets/text/*.h
-	$(RM) -rf build/assets
-	$(RM) -rf .extracted-assets.json
+	$(RM) -r $(ASSET_BIN_DIRS)
+	$(RM) -r $(BUILD_DIR)/assets
+	$(RM) -r assets/text/*.h
+	$(RM) -r .extracted-assets.json
 
 distclean: assetclean clean
-	$(RM) -rf asm baserom data
+	$(RM) -r asm data
+	$(RM) -r baseroms/$(VERSION)/segments
 	$(MAKE) -C tools clean
 
+venv:
+	test -d $(VENV) || python3 -m venv $(VENV)
+	$(PYTHON) -m pip install -U pip
+	$(PYTHON) -m pip install -U -r requirements.txt
+
 ## Extraction step
-setup:
+setup: venv
 	$(MAKE) -C tools
-	$(PYTHON) tools/fixbaserom.py
+	$(PYTHON) tools/decompress_baserom.py $(VERSION)
 	$(PYTHON) tools/extract_baserom.py
-	$(PYTHON) tools/decompress_yars.py
+	$(PYTHON) tools/decompress_yars.py $(VERSION)
 
 assets:
 	$(PYTHON) extract_assets.py -j $(N_THREADS) -Z Wno-hardcoded-pointer
 
 ## Assembly generation
 disasm:
-	$(RM) -rf asm data
+	$(RM) -r asm data
 	$(PYTHON) tools/disasm/disasm.py -j $(N_THREADS) $(DISASM_FLAGS)
 
-diff-init: uncompressed
-	$(RM) -rf expected/
+diff-init: rom
+	$(RM) -r expected/
 	mkdir -p expected/
 	cp -r build expected/build
 
-init:
-	$(MAKE) distclean
+init: distclean
 	$(MAKE) setup
 	$(MAKE) assets
 	$(MAKE) disasm
 	$(MAKE) all
 	$(MAKE) diff-init
```

Following `setup` not implicitly using `venv`, that step would need to be added to the `init` target
```suggestion
init: distclean
	$(MAKE) venv
	$(MAKE) setup
	$(MAKE) assets
	$(MAKE) disasm
	$(MAKE) all
	$(MAKE) diff-init
```